pkg/driver/node.go

```diff
@@ -275,13 +279,16 @@ func (ns *node) NodeUnstageVolume(
 	// immediately other node deleted this node's CR, in that case iSCSI
 	// target(istgt) will pick up the new one and allow only that node to login,
 	// so all the cases are handled
+	utils.TransitionVolListLock.Lock()
 	utils.TransitionVolList[volumeID] = apis.CStorVolumeAttachmentStatusUnmountUnderProgress
 	if err = iscsiutils.UnmountAndDetachDisk(vol, stagingTargetPath); err != nil {
+		utils.TransitionVolListLock.Unlock()
```

Comment:
Since this is a global lock let's unlock this before going into a function which could take time or even hang.